Makefile

```
@@ -431,24 +412,37 @@ define wait
 endef
 
 define generate
-	$(call controller-gen,object:headerFile="hack/boilerplate.go.txt" paths="./...")
+	$(call controller-gen,object:headerFile="hack/boilerplate.gotxt" paths="./...")
 endef
 
 define manifest
 	$(call controller-gen,crd:crdVersions=v1 paths="./..." output:crd:artifacts:config=chart/k8gb/templates/)
 endef
 
-# function retrieves controller-gen path or installs controller-gen@v3.0.0 and retrieve new path in case it is not installed
 define controller-gen
 	@$(if $(filter $(CONTROLLER_GEN_PATH),$(NO_VALUE)),$(call install-controller-gen),)
+	@$(if $(filter-out $(CONTROLLER_GEN_VERSION),$(CONTROLLER_GEN_CURRENT_VERSION)),$(call install-controller-gen),)
 	$(CONTROLLER_GEN_PATH) $1
 endef
 
 define install-controller-gen
-	GO111MODULE=on go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.5.0
+	@echo "\n$(YELLOW)Installing controller-gen $(CYAN)$(CONTROLLER_GEN_VERSION)$(NC)"
+	GO111MODULE=on go install sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_GEN_VERSION)
 	$(eval CONTROLLER_GEN_PATH = $(GOBIN)/controller-gen)
 endef
 
+define golic
+	@$(if $(filter $(GOLIC_PATH),$(NO_VALUE)),$(call install-golic),)
+	@$(if $(filter-out $(GOLIC_VERSION),$(GOLIC_CURRENT_VERSION)),$(call install-golic),)
```

Not sure if we need these conditions. `go install` in 1.16 handles such situation:
 - installs application in case it is missing or different version is present.
 
```suggestion
	@echo "\n$(YELLOW)Installing golic $(CYAN)$(GOLIC_VERSION)$(NC)"
	go install github.com/AbsaOSS/golic@$(GOLIC_VERSION)
```